doc/proposals/tls-utilities.md

```diff
@@ -0,0 +1,159 @@
+# TLS Utilities proposal
+
+## Background:
+
+A typical workflow of an operator that is written using the Operator-SDK involves watching a  Custom Resource Definition (CRD), which specifies the desired state of the cluster, and making changes to the current state of the cluster to achieve the desired state.
+
+Each user specified CRD represents a self contained system. Based on this, we can assume that any TLS assets generated are specific to each system which then is also specific to the corresponding CRD.
+
+Based on the above assumption, we want to design a TLS utility pkg that helps generate TLS assets and ties them to a specific CRD.
+
+## Goals:
+
+To provide TLS utilities for operator developers to create self signed Certificate Authority (CA) and the TLS encryption keys along with the signed certs. This will enable the operator to setup TLS assets for the application so that all the communication within it is secure.
+
+## Details:
+
+We will break down the TLS workflow into the following steps:
+
+1. Generate a Certificate Authority (CA): A CA is a centralized trusted third party that signs the certificates. We need to generate a CA key and CA Certificate to get started. If the users want to provide their own CA, we can add an option to do so. The generated CA certificate can be saved in a ConfigMap as follows:
+	
+	Inputs:
+
+	* CRD-kind
+	* CRD-name
+	* namespace
+
+	Outputs:
+
+        * A ConfigMap that contains the CA Cert:
+
+	```
+	kind: ConfigMap
+	apiVersion: v1
+	metadata:
+  	name: <crd-kind>-<crd-name>-ca
+  	namespace: <ns>
+	data:
+  	ca.pem: ...
```

Comment:
I think we should use `ca.crt` instead of `ca.pem` to keep the naming consistent with the naming scheme `tls.key` and `tls.crt` from the generated TLS secret.